automated_packaging/prepare_major_or_minor_release.pl

```diff
@@ -0,0 +1,154 @@
+# it is time to update the master branch
+system("git checkout $master_branch_name");
+
+$NEW_VERSION_MAJOR=substr($release_major_minor,0,2);
+$NEW_VERSION_MINOR=substr($release_major_minor,2,1) + "1";
+$NEW_VERSION_MAJOR_MINOR=$NEW_VERSION_MAJOR . $NEW_VERSION_MINOR;
+$NEW_VERSION_DEVEL=$NEW_VERSION_MAJOR_MINOR . "devel";
+
+# Update the version on the configuration file
+update_configuration_file($NEW_VERSION_DEVEL, $CURRENT_VERSION_NON_REGEX, "configure.in");
+
+# Run autoconf to generate new configure file
+system("autoconf");
+
+# Update expected version on multi_extension test
+update_configuration_file($NEW_VERSION_DEVEL, $CURRENT_VERSION_NON_REGEX, "./src/test/regress/expected/multi_extension.out");
+
+# We also need to update two different lines on the multi_extension.out
+open(EXTENSION_FILE, "<./src/test/regress/expected/multi_extension.out") || die "File not found";
+my @extension_file_lines = <EXTENSION_FILE>;
+close(EXTENSION_FILE);
+
+my @newlines;
+foreach $line (@extension_file_lines) {
+	$line =~ s/Loaded library requires $CURRENT_VERSION_NON_REGEX_MAJOR_MINOR/Loaded library requires $NEW_VERSION_MAJOR_MINOR/g;
+	push(@newlines,$line);
+}
+
+open(EXTENSION_FILE, ">./src/test/regress/expected/multi_extension.out") || die "File not found";
+print EXTENSION_FILE @newlines;
+close(EXTENSION_FILE);
+
+# Update the Makefile
+open(MAKEFILE, "<./src/backend/distributed/Makefile") || die "File not found";
+my @makefile_lines = <MAKEFILE>;
+close(MAKEFILE);
+
+my @newlines;
+$last_version="";
+foreach $line (@makefile_lines) {
+	$line_copy=$line;
+
+	$line_copy =~ s/^\s+//;
+	$prefix=substr $line_copy, 0, 3;
+	if($prefix eq $release_major_minor) {
+		$line_copy =~ s/\R\z//;
+		@line_parts=split(/ /, $line_copy);
+		$last_version=@line_parts[-1];		
+		$line_copy="	" .  $line_copy . " \\" . "\n	$NEW_VERSION_MAJOR_MINOR-1\n";
+		$line=$line_copy;
+	}
+
+	push(@newlines,$line);
+}
+
+open(MAKEFILE, ">./src/backend/distributed/Makefile") || die "File not found";
+print MAKEFILE @newlines;
+close(MAKEFILE);
+
+# Add a new sql file 
+open(NEW_SQL_FILE, "+>>./src/backend/distributed/citus--$last_version--$NEW_VERSION_MAJOR_MINOR-1.sql") || die "File not found";
+print NEW_SQL_FILE "/* citus--$last_version--$NEW_VERSION_MAJOR_MINOR-1 */ " . "\n\n" . "/* bump version to $NEW_VERSION_MAJOR_MINOR-1 */" . "\n\n";
+close(NEW_SQL_FILE);
+
+# Update citus.control file
+open(CONTROL_FILE, "<./src/backend/distributed/citus.control") || die "File not found";
+my @control_file_lines = <CONTROL_FILE>;
+close(CONTROL_FILE);
+
+my @newlines;
+foreach $line (@control_file_lines) {
+	$line =~ s/$last_version/$NEW_VERSION_MAJOR_MINOR-1/g;
+	push(@newlines,$line);
+}
+
+open(CONTROL_FILE, ">./src/backend/distributed/citus.control") || die "File not found";
+print CONTROL_FILE @newlines;
+close(CONTROL_FILE);
+
+# Push the branch of new release
+`git add ./src/backend/distributed/citus--$last_version--$NEW_VERSION_MAJOR_MINOR-1.sql`;
```

Comment:
I'm not entirely clear why you use backticks sometimes and `system` others… Can you clarify, or unify?